utils/crypto/bls/secret.go

```diff
@@ -19,55 +19,69 @@ var (
 	// The ciphersuite is more commonly known as G2ProofOfPossession.
 	// There are two digests to ensure that message space for normal
 	// signatures and the proof of possession are distinct.
-	ciphersuiteSignature         = []byte("BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
-	ciphersuiteProofOfPossession = []byte("BLS_POP_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
+	ciphersuiteSignature                = []byte("BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
+	ciphersuiteProofOfPossession        = []byte("BLS_POP_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
+	_                            Signer = (*LocalSigner)(nil)
 )
 
 type SecretKey = blst.SecretKey
 
+type Signer interface {
+	PublicKey() *PublicKey
+	Sign(msg []byte) *Signature
+	SignProofOfPossession(msg []byte) *Signature
+	// TODO: delete me
+	ToBytes() []byte
```

Comment:
If we adopt the suggested change on `SecretKeyFromBytes` - then this can be deleted now.
```suggestion
```

Reply:
done fd7f3268a81550f73917cbe36baa1ccef46a639d